Makefile

```diff
@@ -136,54 +131,86 @@ refresh-supervisor-src:
 	&& echo " * Restarting supervisor container.." \
 	&& docker exec -ti resin_supervisor_1 docker restart resin_supervisor
 
-supervisor: gosuper
-	cp Dockerfile.$(DOCKERFILE) Dockerfile
-	echo "ENV VERSION "`jq -r .version package.json` >> Dockerfile
-	echo "ENV DEFAULT_PUBNUB_PUBLISH_KEY $(PUBNUB_PUBLISH_KEY)" >> Dockerfile
-	echo "ENV DEFAULT_PUBNUB_SUBSCRIBE_KEY $(PUBNUB_SUBSCRIBE_KEY)" >> Dockerfile
-	echo "ENV DEFAULT_MIXPANEL_TOKEN $(MIXPANEL_TOKEN)" >> Dockerfile
+supervisor: nodesuper gosuper
+	sed 's/%%ARCH%%/$(ARCH)/g' Dockerfile.runtime.template > Dockerfile.runtime.$(ARCH)
+	echo "ENV VERSION $(shell jq -r .version package.json) \\" >> Dockerfile.runtime.$(ARCH)
```

Comment:
Well, with build args I'd have to put the following in the Dockerfile and still handle it in the Makefile. For the purposes of minimising size and layer cound I think I'll leave it here

```Dockerfile
ARG VERSION
ENV VERSION $VERSION
```